src/anyp/Uri.h

```diff
@@ -98,12 +100,24 @@ class Uri
     void path(const SBuf &p) {path_=p; touch();}
     const SBuf &path() const;
 
+    /**
+     * Merge a relative-path URL into the existing URI details.
+     * Implements RFC 3986 section 5.2.3
+     *
+     * The caller must ensure relUrl is a valid relative-path.
```

Comment:
AFAICT, this part of the description does not match the code. The [RFC](https://tools.ietf.org/html/rfc3986#section-5.2.3) defines relative-path as a "relative reference that does not begin with a slash character". The new addRelativePath() method has regular-looking code for adding relUrl strings that start with a slash character. Do we need to add a comment to that code that would blame the caller for supplying an invalid parameter or should we relax this part of the interface documentation?

Reply:
You mean s/must/should/  ?

Reply:
> You mean s/must/should/ ?

No, replacing "must" with "should" would not address the mismatch I described. I see two ways to solve this problem:

1. Add a `FIXME` comment to the "adding relUrl strings that start with a slash character" method code that would blame the caller for supplying an invalid parameter. The interface documentation stays the same. The code is just tolerant to gross API violations. The callers should be eventually fixed.

2. Relax this part of the interface documentation so that a relUrl strings that start with a slash character become first-class citizens rather than API violations. The parameter should be renamed in this case as well.

The choice is yours -- you know the corresponding use cases (that supply the parameter) better than I do.